k8s.go

```diff
@@ -0,0 +1,247 @@
+package profitbricks
+
+import "net/http"
+
+type KubernetesClusters struct {
+	// URL to the collection representation (absolute path)
+	// Read Only: true
+	// Format: uri
+	Href string `json:"href,omitempty"`
+
+	// Unique representation for Kubernetes Cluster as a collection on a resource.
+	// Read Only: true
+	ID string `json:"id,omitempty"`
+
+	// Array of items in that collection
+	// Read Only: true
+	Items []*KubernetesCluster `json:"items"`
+
+	// The type of resource within a collection
+	// Read Only: true
+	// Enum: [k8s]
+	PBType string `json:"type,omitempty"`
+}
+
+// KubernetesCluster kubernetes cluster
+type KubernetesCluster struct {
+	// URL to the object representation (absolute path)
+	// Read Only: true
+	// Format: uri
+	Href string `json:"href,omitempty"`
+
+	// The resource's unique identifier.
+	// Read Only: true
+	ID string `json:"id,omitempty"`
+
+	// metadata
+	Metadata *NoStateMetaData `json:"metadata,omitempty"`
+
+	// properties
+	// Required: true
+	Properties *KubernetesClusterProperties `json:"properties"`
+
+	// The type of object
+	// Read Only: true
+	// Enum: [k8s]
+	PBType string `json:"type,omitempty"`
+
+	// Entities of a cluster
+	Entities KubernetesClusterEntities `json:"entities"`
+}
+
+type KubernetesClusterEntities struct {
+	// NodePools of a cluster
+	NodePools *KubernetesNodePools `json:"nodepools,omitempty"`
```

Comment:
`KubernetesNodePools` has `Items`. According to https://devops.ionos.com/api/cloud/v5/#read-cluster this is not the case for a cluster's nodepool entities.

Reply:
Probably the example in the doc was not with increased depth. Please check the examples from testdata, they are based on real life responses